.github/workflows/python-test-pypi-package.yml

```diff
@@ -0,0 +1,75 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Test & Upload to TestPyPI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push to the test branch
+  push:
```

Comment:
Just to double check, does PUSH include raising a PR?

Reply:
I missed that; there is a separate event for `on: pull_request: [closed]`, which is followed by an `if_merged` job in the examples. I'm looking into this, and trying to figure out how to run the job if the `test` branch is target of the pull request.

Source : [GitHub Actions/Using workflows/Events that trigger workflows/Running your workflow when a pull request merges](https://docs.github.com/en/github-ae@latest/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges)